Source/Dafny/DafnyOptions.cs

```diff
@@ -85,7 +104,7 @@ public enum PrintModes {
     public bool DafnyVerify = true;
     public string DafnyPrintResolvedFile = null;
     public List<string> DafnyPrintExportedViews = new List<string>();
-    public bool Compile = true;
+    public bool EmitExecutable = true;
```

Comment:
If we emit a library, what will be the option for that?

Reply:
> `Emit` suggests to me that an executable is going to be written out to disk. Why the rename?

Doesn't `Compile` mean that a binary will be written to disk?

Reply:
> Doesn't Compile mean that a binary will be written to disk?

Yes, but in Dafny's case it usually mean emit the translated source files to disk.

I think you want `EmitBinary` rather than `EmitExecutable` here, since you should also be able to emit libraries which are not executable.

Reply:
> Doesn't Compile mean that a binary will be written to disk?

Sorry, I wasn't clear.  AFAICT, `Compile = true` currently doesn't mean that anything will be written to disk.  This may not be good terminology (I think it's fine).  In this PR, `Compile` is replaced by `EmitExecutable`, but the behavior is unchanged (there's still not necessarily something being written).  Given this, does the rename make sense?  This flag doesn't indicate that anything will actually be emitted, right?

Reply:
> AFAICT, Compile = true currently doesn't mean that anything will be written to disk

So, what does it do then? I tried to decipher the code and it does seem to achieve this, given that the program verified correctly. Does it not?

Reply:
In some languages we don't need to write anything to disk, for example in C#: we compile in memory.